lock-api/src/main/java/com/palantir/lock/v2/TimestampRangeAndPartition.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.v2;
+
+import org.immutables.value.Value;
+
+import com.palantir.timestamp.TimestampRange;
+
+@Value.Immutable
+public interface TimestampRangeAndPartition {
```

Comment:
I'd like to follow up on this; I think that this class will only really make sense if it's documented correctly both in explicit comments and actually the code.

So, instead of returning a range and a modulus indicating which in the range are valid mod 16, maybe we should instead be returning:

1. Start
2. Interval
3. Count

indicating the beginning timestamp, the difference between each timestamp, and the number of timestamps returned?

In Jeremy's case, you'd then be returning(0, 16, 2) which is to me much more fully specified?

Reply:
it also means the client doesn't have to know about how many partitions we have, which is also much nicer; it means we can change it down the line and don't need SharedConstants.

Reply:
Yep, I think I prefer this specific triple format over trying to reuse `TimestampRange`